crates/ruff_linter/src/rules/pylint/rules/no_method_decorator.rs

```diff
@@ -131,7 +131,7 @@ fn get_undecorated_methods(
 
                         if let Expr::Name(ast::ExprName { id, .. }) = &arguments.args[0] {
                             if target_name == *id {
-                                explicit_decorator_calls.insert(id.clone(), stmt.range());
```

Comment:
GitHub doesn't allow me to make a suggestion on the entire code. We can avoid calling `to_string` here by changing `explicit_decorator_calls` to `HashMap<&str, TextRange>`. 

The `id.to_string` call on linne 128 also seems unnecessary (we can use the `SmolStr` directly or use `as_str`)